metadata-service/auth-impl/src/main/java/com/datahub/authentication/authenticator/DatahubJwtTokenAuthenticator.java

```diff
@@ -0,0 +1,121 @@
+package com.datahub.authentication.authenticator;
+
+import com.auth0.jwk.JwkException;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwk.UrlJwkProvider;
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.Claim;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.datahub.authentication.Actor;
+import com.datahub.authentication.ActorType;
+import com.datahub.authentication.Authentication;
+import com.datahub.authentication.AuthenticationException;
+import com.datahub.authentication.AuthenticationRequest;
+import com.datahub.plugins.auth.authentication.Authenticator;
+import com.datahub.authentication.AuthenticatorContext;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.interfaces.RSAPublicKey;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.datahub.authentication.AuthenticationConstants.*;
+
+
+/**
+ * This Authenticator verifies third party token and extract custom claim for userId.
+ * Supported algorithm at this moment RS256
+ */
+@Slf4j
+public class DatahubJwtTokenAuthenticator implements Authenticator {
+
+
+  /**
+   *  Custom claim name, configured to extract claim from token and set as user Id
+   * **/
+  private String idClaim;
+
+  /**
+   *  List of trusted issuers
+   * **/
+  private LinkedHashMap<String, String> trustedIssuers;
+
+  @Override
+  public void init(@Nonnull final Map<String, Object> config, @Nullable final AuthenticatorContext context) {
+    Objects.requireNonNull(config, "Config parameter cannot be null");
+    this.idClaim = Objects.requireNonNull((String) config.get("idClaim"), "Missing required config Claim Name");
+
+    this.trustedIssuers = Objects.requireNonNull((LinkedHashMap<String, String>) config.get("trustedIssuers"),
```

Comment:
This seems dangerous - We cannot assume that the underlying data is actually of type "LinkedHashMap"

Reply:
This is default conversion of list  in application.yml, with indexes used as key and and list item as values. The other thing i can try is to use Map<String,String> for convertion and then map to Set<String>. Let me know ur thoughts.